backend/src/main/java/com/jobwait/security/exchange/LinkedInOAuthExchange.java

```diff
@@ -0,0 +1,68 @@
+package com.jobwait.security.exchange;
+
+import java.util.UUID;
+
+import org.springframework.http.MediaType;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import com.jobwait.security.AuthToken;
+import com.jobwait.security.SecurityFaults;
+import com.jobwait.security.UUIDJumbler;
+import com.jobwait.security.linkedin.LinkedInUserProfile;
+import com.jobwait.util.Tuple;
+
+public final class LinkedInOAuthExchange extends OAuthExchange {
+    private static WebClient localClient = WebClient.create();
+
+    private static String codeToBearerToken(String authCode) {
+        // Everything except authCode should be pulled in from ENV
+        final String clientId = "";
+        final String clientSecret = "";
+        final String redirectURI = "";
+        final String linkedInAccessTokenURI = "https://www.linkedin.com/oauth/v2/accessToken";
+
+        MultiValueMap<String, String> formData = new LinkedMultiValueMap<>();
+        formData.add("grant_type", "authorization_code");
+        formData.add("code", authCode);
+        formData.add("client_id", clientId);
+        formData.add("client_secret", clientSecret);
+        formData.add("redirect_uri", redirectURI);
+
+        return localClient
+                .post()
+                .uri(linkedInAccessTokenURI)
+                .contentType(MediaType.APPLICATION_FORM_URLENCODED)
+                .bodyValue(formData)
+                .retrieve()
+                .bodyToMono(AuthToken.class)
+                .onErrorMap(err -> SecurityFaults.FailToGetTokenFault(err.getMessage()))
+                .block().authToken;
```

Comment:
So will this throw a SecurityFault if there's a problem? Or will it return undefined for the `.authToken` part?

Reply:
Absolutely fantastic question which I cannot answer because I have not tested it. I would assume it either throws an error or something equivalent to a `Promise.reject(error)` in JS.

Reply:
Hopefully (and most likely) it's an error, so that should be caught by the error convertor at the top of the architecture. I'm sure this will come up in testing so I'm not too concerned, mostly just curious